Loop/Managers/DeviceDataManager.swift

```diff
@@ -1372,156 +1388,144 @@ extension DeviceDataManager {
+            } catch {
+                log.error("Remote Notification: Override Action Cancel Error: %{public}@", String(describing: error))
             }
-            
-            guard candidateCarbsInGrams <= LoopConstants.maxCarbEntryQuantity.doubleValue(for: .gram()) else {
-                NotificationManager.sendRemoteCarbEntryFailureNotification(for: RemoteCommandError.exceedsMaxCarbs, amountInGrams: candidateCarbsInGrams)
-                log.error("Remote Notification: Carbs higher than maximum. Aborting...")
-                return
+        case .bolusEntry(let bolusAction):
+            do {
+                try validatePushNotificationSource(notification: notification)
+                try await handleRemoteBolusAction(bolusAction)
+            } catch {
+                await NotificationManager.sendRemoteBolusFailureNotification(for: error, amount: bolusAction.amountInUnits)
+                log.error("Remote Notification: Bolus Action Error: %{public}@", String(describing: notification))
             }
-
-            addRemoteCarbEntry(candidateCarbEntry) { carbEntryAddResult in
-                switch carbEntryAddResult {
-                case .success(let completedCarbEntry):
-                    NotificationManager.sendRemoteCarbEntryNotification(amountInGrams: completedCarbEntry.quantity.doubleValue(for: .gram()))
-                case .failure(let error):
-                    self.log.error("Remote Notification: Error adding carb entry: %{public}@", String(describing: error))
-                    NotificationManager.sendRemoteCarbEntryFailureNotification(for: error, amountInGrams: candidateCarbsInGrams)
-                }
+        case .carbsEntry(let carbAction):
+            do {
+                try validatePushNotificationSource(notification: notification)
+                try await handleRemoteCarbAction(carbAction)
+            } catch {
+                await NotificationManager.sendRemoteCarbEntryFailureNotification(for: error, amountInGrams: carbAction.amountInGrams)
+                log.error("Remote Notification: Carb Action Error: %{public}@", String(describing: notification))
             }
         }
     }
     
-    func activateRemoteOverride(_ remoteOverride: TemporaryScheduleOverride?) {
+    func validatePushNotificationSource(notification: [String: AnyObject]) throws {
+        
+        let defaultServiceIdentifier = "NightscoutService"
+        let serviceIdentifer = notification["serviceIdentifier"] as? String ?? defaultServiceIdentifier
+        
+        let validationResult = remoteDataServicesManager.validatePushNotificationSource(notification, serviceIdentifier: serviceIdentifer)
+        switch validationResult {
+        case .success():
+            log.info("Remote Notification: Validation successful")
+        case .failure(let error):
+            throw error
+        }
+    }
+    
+    //Remote Overrides
+    
```

Comment:
The paradigm here is each RemoteAction is asked for its data required to enact it (i.e. TemporaryOverride, CarbEntry, etc). 

That call allows the action to validate that its data is normalized wrt to Loop's expectations (ex: carbs not exceeding reasonable max)